src/main/java/org/opensearch/performanceanalyzer/collectors/telemetry/RTFDisksCollector.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.performanceanalyzer.collectors.telemetry;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.performanceanalyzer.OpenSearchResources;
+import org.opensearch.performanceanalyzer.commons.OSMetricsGeneratorFactory;
+import org.opensearch.performanceanalyzer.commons.collectors.PerformanceAnalyzerMetricsCollector;
+import org.opensearch.performanceanalyzer.commons.collectors.TelemetryCollector;
+import org.opensearch.performanceanalyzer.commons.config.overrides.ConfigOverridesWrapper;
+import org.opensearch.performanceanalyzer.commons.metrics.MetricsConfiguration;
+import org.opensearch.performanceanalyzer.commons.metrics.RTFMetrics;
+import org.opensearch.performanceanalyzer.commons.metrics_generator.DiskMetricsGenerator;
+import org.opensearch.performanceanalyzer.commons.metrics_generator.OSMetricsGenerator;
+import org.opensearch.performanceanalyzer.commons.stats.metrics.StatExceptionCode;
+import org.opensearch.performanceanalyzer.commons.stats.metrics.StatMetrics;
+import org.opensearch.performanceanalyzer.config.PerformanceAnalyzerController;
+import org.opensearch.telemetry.metrics.Histogram;
+import org.opensearch.telemetry.metrics.MetricsRegistry;
+import org.opensearch.telemetry.metrics.tags.Tags;
+
+public class RTFDisksCollector extends PerformanceAnalyzerMetricsCollector
+        implements TelemetryCollector {
+
+    private Histogram diskWaitTimeMetrics;
+    private Histogram diskServiceRateMetrics;
+    private Histogram diskUtilizationMetrics;
+    private MetricsRegistry metricsRegistry;
+    private boolean metricsInitialised;
+    private static final Logger LOG = LogManager.getLogger(RTFDisksCollector.class);
+    private PerformanceAnalyzerController performanceAnalyzerController;
+    private ConfigOverridesWrapper configOverridesWrapper;
+
+    public RTFDisksCollector(
+            PerformanceAnalyzerController performanceAnalyzerController,
+            ConfigOverridesWrapper configOverridesWrapper) {
+        super(
+                MetricsConfiguration.CONFIG_MAP.get(RTFDisksCollector.class).samplingInterval,
+                "RTFDisksCollector",
+                StatMetrics.DISKS_COLLECTOR_EXECUTION_TIME,
```

Comment:
Why are we publishing the stats against the existing RCA collector metrics?

Reply:
This is for PMET. I can update it but it's not required since it is dependent on RCA from what I understand. We need to look into that statsCollector but I would still update this one for consistency.